tools/python/clean_strings_txt.py

```diff
@@ -0,0 +1,330 @@
+    strings_txt = StringsTxt()
+    strings_dict = {key : dict(strings_txt.translations[key]) for key in filtered}
+    strings_txt.translations = strings_dict
+    strings_txt.comments_and_tags = {}
+    strings_txt.write_formatted(filepath, languages=languages)
+
+
+def get_args():
+    parser = ArgumentParser(
+        description="""
+        A script for cleaning up the strings.txt file. It can cleanup the file
+        inplace, that is all the unused strings will be removed from strings.txt,
+        or it can produce two separate files for ios and android. We can also
+        produce the compiled string resources specifically for each platform
+        that do not contain strings for other platforms or comments."""
+    )
+
+    parser.add_argument(
+        "-s", "--single-file",
+        dest="single", default=False,
+        action="store_true",
+        help="""Create single cleaned up file for both platform. All strings
+        that are not used in the project will be thrown away. Otherwise, two
+        platform specific files will be produced."""
+    )
+
+    parser.add_argument(
+        "-l", "--language",
+        dest="langs", default=None,
+        action="append",
+        help="""The languages to be included into the resulting strings.txt
+        file or files. If this param is empty, all languages from the current
+        strings.txt file will be preserved."""
+    )
+
+    parser.add_argument(
+        "-g", "--generate-localizations",
+        dest="generate", default=False,
+        action="store_true",
+        help="Generate localizations for the platforms."
+    )
+
+    parser.add_argument(
+        "-o", "--output",
+        dest="output", default="strings.txt",
+        help="""The name for the resulting file. It will be saved to the
+        project folder. Only relevant if the -s option is set."""
+    )
+
+    parser.add_argument(
+        "-m", "--missing-strings",
+        dest="missing", default=False,
+        action="store_true",
+        help="""Find the keys that are used in iOS, but are not translated
+        in strings.txt and exit."""
+    )
+
+    parser.add_argument(
+        "-c", "--candidates",
+        dest="candidates", default=False,
+        action="store_true",
+        help="""Find the strings in iOS that are not in the L() macros, but that
+        look like they might be keys."""
+    )
+
+    parser.add_argument(
+        "-r", "--root",
+        dest="omim_root", default=None,
+        help="Path to the root of the OMIM project"
+    )
+
+    parser.add_argument(
+        "-ct", "--categories",
+        dest="hardcoded_cagegories", default=None,
+        help="""Path to the list of the categories that are displayed in the
+        interface, but are not taken from strings.txt"""
+    )
+
+    return parser.parse_args()
+
+
+def do_multiple(args):
+    write_filtered_strings_txt(
+        grep_ios(), "{0}/ios_strings.txt".format(OMIM_ROOT), args.langs
```

Comment:
We decided not to do os.path.join, because in that case we would have to to do it to all paths with all their components. Format is a more convenient way of doing this, as we might want to set other parts of the path later.